roles/etcd/templates/etcd.service.j2

```diff
@@ -0,0 +1,38 @@
+[Unit]
+Description=etcd
+After=docker.service
+Requires=docker.service
+
+[Service]
+Restart=on-failure
+RestartSec=20
+TimeoutStartSec=20m
+
+ExecStartPre=-/usr/bin/docker kill etcd
+ExecStartPre=-/usr/bin/docker rm etcd
+ExecStartPre=/usr/bin/docker pull {{ etcd_image }}:{{ etcd_image_tag }}
+
+ExecStart=/usr/bin/docker run \
+    --name=etcd \
+    --publish={{ etcd_client_port }}:{{ etcd_client_port }} \
+    --publish={{ etcd_peer_port }}:{{ etcd_peer_port }} \
+    {{ etcd_image }}:{{ etcd_image_tag }} \
+    --name {{ inventory_hostname }} \
+    --listen-peer-urls http://0.0.0.0:{{ etcd_peer_port }} \
+    --listen-client-urls http://0.0.0.0:{{ etcd_client_port }} \
+    --initial-advertise-peer-urls http://{{ inventory_hostname }}:{{ etcd_peer_port }} \
+    --initial-cluster-state new \
+    --advertise-client-urls http://{{ inventory_hostname }}:{{ etcd_client_port }} \
+{# Generate string like: 'member1=http://host1:port,member2=http://host2:port' #}
+    --initial-cluster {% for item in groups['all'] -%}
```

Comment:
I like this idea!

Sent from a mobile device.

On 28 Jul 2015, at 19:08, "Steven Borrelli" <notifications@github.com<mailto:notifications@github.com>> wrote:


In roles/etcd/templates/etcd.service.j2<https://github.com/CiscoCloud/microservices-infrastructure/pull/598#discussion_r35679802>:

> +ExecStartPre=-/usr/bin/docker rm etcd
> +ExecStartPre=/usr/bin/docker pull {{ etcd_image }}:{{ etcd_image_tag }}
> +
> +ExecStart=/usr/bin/docker run \
> +    --name=etcd \
> +    --publish={{ etcd_client_port }}:{{ etcd_client_port }} \
> +    --publish={{ etcd_peer_port }}:{{ etcd_peer_port }} \
> +    {{ etcd_image }}:{{ etcd_image_tag }} \
> +    --name {{ inventory_hostname }} \
> +    --listen-peer-urls http://0.0.0.0:{{ etcd_peer_port }} \
> +    --listen-client-urls http://0.0.0.0:{{ etcd_client_port }} \
> +    --initial-advertise-peer-urls http://{{ inventory_hostname }}:{{ etcd_peer_port }} \
> +    --initial-cluster-state new \
> +    --advertise-client-urls http://{{ inventory_hostname }}:{{ etcd_client_port }} \
> +{# Generate string like: 'member1=http://host1:port,member2=http://host2:port' #}
> +    --initial-cluster {% for item in groups['all'] -%}


Could we use etcd SRV discovery using consul dns for this instead of using ansible vars?
https://github.com/coreos/etcd/blob/master/Documentation/clustering.md#dns-discovery

—
Reply to this email directly or view it on GitHub<https://github.com/CiscoCloud/microservices-infrastructure/pull/598/files#r35679802>.